pkg/controller/jobframework/reconciler.go

```diff
@@ -109,9 +110,21 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 
 	// when manageJobsWithoutQueueName is disabled we only reconcile jobs that have either
 	// queue-name or the parent-workload annotation set.
-	if !r.manageJobsWithoutQueueName && QueueName(job) == "" && isStandaloneJob {
-		log.V(3).Info(fmt.Sprintf("Neither %s label, nor %s annotation is set, ignoring the job", QueueLabel, ParentWorkloadAnnotation))
-		return ctrl.Result{}, nil
+	// If the parent-workload annotation is set, it also checks whether the parent job has queue-name label.
+	if !r.manageJobsWithoutQueueName && QueueName(job) == "" {
+		if isStandaloneJob {
+			log.V(3).Info(fmt.Sprintf("Neither %s label, nor %s annotation is set, ignoring the job", controllerconsts.QueueLabel, controllerconsts.ParentWorkloadAnnotation))
```

Comment:
Do not use `Sprintf` for logging. Instead, use a constant message and add keys and values, so we get structured logging.

Reply:
It makes sense.

Reply:
done.